cli/src/commands/run.rs

```diff
@@ -24,24 +30,41 @@ use crate::ui::Ui;
 /// # Example
 ///
 /// # Run pre-commit on your local work
-/// $ jj run 'pre-commit.py .github/pre-commit.yaml' -r (main..@) -j 4
+/// $ jj run 'pre-commit run .github/pre-commit.yaml' -r (trunk()..@) -j 4
 ///
 /// This allows pre-commit integration and other funny stuff.
 #[derive(clap::Args, Clone, Debug)]
 #[command(verbatim_doc_comment)]
-pub(crate) struct RunArgs {
+pub struct RunArgs {
     /// The command to run across all selected revisions.
-    #[arg(long, short, alias = "x")]
-    command: String,
+    shell_command: String,
     /// The revisions to change.
+    /// Multiple revsets are accepted and the work will be done on a
+    /// intersection of them.
```

Comment:
I'm not sure if I used the wrong term but the thing I want to express is `jj run '<command> -r '<revset1>' -r '<revset2>'` will only use the commits which are in both revsets.

Reply:
I think that you've written what you want to express, but why would you want the intersection in that case rather than the union? Is there a particular workflow that you have in mind? Are there other cases of jj commands accepting multiple `-r`s, and do they calculate the intersection or the union of those sets?

Reply:
> but why would you want the intersection in that case rather than the union? Is there a particular workflow that you have in mind?

Basically I want to minimize work, while giving the user enough space to express themselves correctly with revsets. 
It acts as a crutch (for people like me) which aren't precise with their revsets. 

> Are there other cases of jj commands accepting multiple `-r`s, and do they calculate the intersection or the union of those sets?

Yes, but none of them calculate a intersection yet. Only `jj log` takes multiple revsets and unions them.

Reply:
FWIW, I would also expect it to union the revsets like I think `jj log`, `jj abandon`, `jj duplicate` and `jj git push` do. Actually it looks like `jj run` also does right now:
https://github.com/martinvonz/jj/blob/7d9223d7343a14f22fe855ef33dc59887c3dff08/cli/src/commands/run.rs#L56-L57

Reply:
Addressed by #2553